CHANGELOG.md

```diff
@@ -45,6 +45,7 @@
 * [BUGFIX] Blocks storage: do not cleanup a partially uploaded block when `meta.json` upload fails. Despite failure to upload `meta.json`, this file may in some cases still appear in the bucket later. By skipping early cleanup, we avoid having corrupted blocks in the storage. #3660
 * [BUGFIX] Alertmanager: disable access to `/alertmanager/metrics` (which exposes all Cortex metrics), `/alertmanager/-/reload` and `/alertmanager/debug/*`, which were available to any authenticated user with enabled AlertManager. #3678
 * [BUGFIX] Query-Frontend: avoid creating many small sub-queries by discarding cache extents under 5 minutes #3653
+* [BUGFIX] Ruler: Minimize gaps on rule evaluations with stale input and enabled ruler evaluation delay. #3687
```

Comment:
**suggestion:** I think it would be good to make it clear what was fixed in addition to the symptom is solves.
```suggestion
* [BUGFIX] Ruler: Ensure the stale markers generated for evaluated rules respect the set evaluation delay. This will avoid issues with samples with NaN be persisted with timestamps set ahead of the next rule evaluation. #3687
```